plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -269,55 +296,79 @@ function OPDSBrowser:fetchFeed(item_url, username, password, method)
         password = password,
     }
     logger.info("Request:", request)
+    -- Fire off the request and wait to see what we get back.
     local code, headers = socket.skip(1, http.request(request))
     socketutil:reset_timeout()
-    -- raise error message when network is unavailable
+    -- Check the response and raise error message when network is unavailable.
     if headers == nil then
         error(code)
     end
+    -- Below are numerous if cases to handle different response codes.
     if code == 200 then
+        -- 200 means the request succeeded.
+        -- If the method sent was HEAD, then we're probably checking for
+        -- an update and therefore only interested in the last-modified
+        -- time of the resource (who needs a body when you have a head?).
         if method == "HEAD" then
             if headers["last-modified"] then
                 return headers["last-modified"]
             else
                 return
             end
         end
+        -- If the method sent was not HEAD, then we are interested in
+        -- the payload of the request. We'll add that to a table below
+        -- and return that as the result of this function.
         local xml = table.concat(sink)
+        -- Obviously, check to see if the payload exists.
         if xml ~= "" then
             return xml
         end
     elseif method == "HEAD" then
         -- Don't show error messages when we check headers only.
         return
-    elseif code == 301 then
-        UIManager:show(InfoMessage:new{
-            text = T(_("The catalog has been permanently moved. Please update catalog URL to '%1'."), BD.url(headers['Location'])),
+    elseif code == 301 then -- Page has permanently moved
+        UIManager:show(
+            InfoMessage:new{
+                text = T(_("The catalog has been permanently moved. Please update catalog URL to '%1'."),
+                         BD.url(headers['Location'])),
         })
-    elseif code == 302 and item_url:match("^https") and headers.location:match("^http[^s]") then
-        UIManager:show(InfoMessage:new{
-            text = T(_("Insecure HTTPS → HTTP downgrade attempted by redirect from:\n\n'%1'\n\nto\n\n'%2'.\n\nPlease inform the server administrator that many clients disallow this because it could be a downgrade attack."), BD.url(item_url), BD.url(headers.location)),
-            icon = "notice-warning",
+    elseif code == 302
+        and item_url:match("^https")
+        and headers.location:match("^http[^s]") then -- Page is redirecting
+        UIManager:show(
+            InfoMessage:new{
+                text = T(_("Insecure HTTPS → HTTP downgrade attempted by redirect from:\n\n'%1'\n\nto\n\n'%2'.\n\nPlease inform the server administrator that many clients disallow this because it could be a downgrade attack."),
+                         BD.url(item_url),
+                         BD.url(headers.location)),
+                icon = "notice-warning",
```

Comment:
So could you make them all like this? Pinging @poire-z for confirmation that we're on the same page.
```suggestion
        UIManager:show(InfoMessage:new{
            text = T(_("Insecure HTTPS → HTTP downgrade attempted by redirect from:\n\n'%1'\n\nto\n\n'%2'.\n\nPlease inform the server administrator that many clients disallow this because it could be a downgrade attack."),
                     BD.url(item_url),
                     BD.url(headers.location)),
            icon = "notice-warning",
```

Reply:
That looks fine to me. It's not my preference but I honestly don't mind adhering to the conventions. I guess I just haven't learned what they _all_ are yet, so that's why I'm introducing my own.

Reply:
> that we're on the same page.

Looks fine to me too.